WriteHDLUtils.py

```diff
@@ -818,23 +882,44 @@ def writeModuleInst_generic(module, hls_src_dir, f_writeTemplatePars,
                         # Temporary bodge to account for encoded index in fullmatch memories
                         if tmp_argname == 'fullmatch':
                             tmp_argname += "_" + str(decodeSeedIndex_MC(memory.inst))
+                        elif tmp_argname == 'projout_barrel_ps' or tmp_argname == 'projout_barrel_2s' or tmp_argname == 'projout_disk':
+                            tmp_argname += "_" + str(decodeSeedIndex_TC(memory.inst))
                         else:
                             tmp_argname += "_" + str(array_dict[tmp_argname])
-
+                    
                     # Add the memory instance to the port string
                     # Assumes a sorted memModuleList due to arrays?
-                    if portname.find("in") != -1:
-                        string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,memory)
-                    if portname.find("out") != -1:
-                        string_mem_ports += writeProcMemoryLHSPorts(tmp_argname,memory)
+                    if portname.replace("inner","").find("in") != -1:
+                        if isinstance(memory, list):
+                            for memmodule in memory:
+                                string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,  memmodule)
+                        else:
+                            string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,memory)
+                    if portname.replace("outer","").find("out") != -1:
+                        if isinstance(memory, list):
+                            for memmodule in memory:
+                                string_mem_ports += writeProcMemoryLHSPorts(tmp_argname,memmodule)
+                        else:
+                            string_mem_ports += writeProcMemoryLHSPorts(tmp_argname,memory)
+                    if portname.find("trackpar") != -1 and module.mtype == "TrackletCalculator":
+                        if isinstance(memory, list):
+                            for memmodule in memory:
+                                string_mem_ports += writeProcMemoryLHSPorts(tmp_argname,  memmodule)
+                        else:
+                            string_mem_ports += writeProcMemoryLHSPorts(tmp_argname,memory)
+                    elif portname.find("trackpar") != -1 and module.mtype == "PurgeDuplicates":
+                        if isinstance(memory, list):
+                            for memmodule in memory:
+                                string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,  memmodule)
+                        else:
+                            string_mem_ports += writeProcMemoryRHSPorts(tmp_argname,memory)
```

Comment:
Why do we have all of this? Is memory ever a list?

Reply:
I am not sure why this was added in. You are right that it never seems to be a list as far as I can tell. I've removed all the checks in most recent commit.